encoder-otel-brave/src/main/java/zipkin2/reporter/otel/brave/SpanTranslator.java

```diff
@@ -0,0 +1,215 @@
+   *
+   * <p>Ex.
+   *
+   * <pre>{@code
+   * tracesData = SpanTranslator.translate(braveSpan);
+   * }</pre>
+   *
+   * @param braveSpan The Zipkin Span.
+   * @return A OpenTelemetry Span.
+   */
+  TracesData translate(MutableSpan braveSpan) {
+    TracesData.Builder tracesDataBuilder = TracesData.newBuilder();
+    Builder resourceSpansBuilder = ResourceSpans.newBuilder();
+    ScopeSpans.Builder scopeSpanBuilder = ScopeSpans.newBuilder();
+    Span.Builder spanBuilder = builderForSingleSpan(braveSpan, resourceSpansBuilder);
+    scopeSpanBuilder.addSpans(spanBuilder
+        .build());
+    InstrumentationScope.Builder scopeBuilder = InstrumentationScope.newBuilder();
+    scopeBuilder.setName(BraveScope.getName());
+    scopeBuilder.setVersion(BraveScope.getVersion());
+    scopeSpanBuilder.setScope(scopeBuilder.build());
+    resourceSpansBuilder.addScopeSpans(scopeSpanBuilder
+        .build());
+    tracesDataBuilder.addResourceSpans(resourceSpansBuilder.build());
+    return tracesDataBuilder.build();
+  }
+
+  static KeyValue stringAttribute(String key, String value) {
+    return KeyValue.newBuilder()
+        .setKey(key)
+        .setValue(AnyValue.newBuilder().setStringValue(value))
+        .build();
+  }
+
+  static KeyValue intAttribute(String key, int value) {
+    return KeyValue.newBuilder()
+        .setKey(key)
+        .setValue(AnyValue.newBuilder().setIntValue(value))
+        .build();
+  }
+
+  private Span.Builder builderForSingleSpan(MutableSpan span, Builder resourceSpansBuilder) {
+    Span.Builder spanBuilder = Span.newBuilder()
+        .setTraceId(ByteString.fromHex(span.traceId() != null ? span.traceId() : io.opentelemetry.api.trace.SpanContext.getInvalid().getTraceId()))
+        .setSpanId(ByteString.fromHex(span.id() != null ? span.id() : io.opentelemetry.api.trace.SpanContext.getInvalid().getSpanId()))
+        .setName((span.name() == null || span.name().isEmpty()) ? "unknown" : span.name());
+    if (span.parentId() != null) {
+      spanBuilder.setParentSpanId(ByteString.fromHex(span.parentId()));
+    }
+    long start = span.startTimestamp();
+    long finish = span.finishTimestamp();
+    spanBuilder.setStartTimeUnixNano(TimeUnit.MICROSECONDS.toNanos(start));
+    if (start != 0 && finish != 0L) {
+      spanBuilder.setEndTimeUnixNano(TimeUnit.MICROSECONDS.toNanos(finish));
+    }
+    Kind kind = span.kind();
+    if (kind != null) {
+      switch (kind) {
+        case CLIENT:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_CLIENT);
+          break;
+        case SERVER:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_SERVER);
+          break;
+        case PRODUCER:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_PRODUCER);
+          break;
+        case CONSUMER:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_CONSUMER);
+          break;
+        default:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_INTERNAL); //TODO: Should it work like this?
```

Comment:
Since the spec does not state that setting kind is required, and Brave has no plans to add SpanKind, I have removed the "default" case in https://github.com/openzipkin-contrib/zipkin-otel/pull/9/commits/4b785669b1d55c6a0057d94985424aac36bf92ed